### PR TITLE
Changes made based on feedback

### PR DESCRIPTION
**Pull request by raahi-chada2004**

### CODE REVIEW

File: out_writeapi_test.go [hunk 1]
```diff
@@ -38,3 +56,150 @@ func TestFLBPluginRegister(t *testing.T) {
...
+}
+
+func (m *MockManagedWriterClient) CreateWriteStream(ctx context.Context, req *storagepb.CreateWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+	return m.client.CreateWriteStream(ctx, req, opts...)
+}
+
+// TestFLBPluginInit tests the FLBPluginInit function
+func TestFLBPluginInit(t *testing.T) {
+	//variable keeping track of what order functions are called in
+	order := 0
```
Comment (agrawal-siddharth) [inline out_writeapi_test.go]: Checking for the order in which the functions are called seems excessive and verging on implementation-dependent. We generally don't do that. Verifying that each function is called the expected number of times is sufficient.

**Review by agrawal-siddharth — CHANGES_REQUESTED** [review]
Can changes to app/logfile.log be omitted?

File: out_writeapi_test.go (hunk 1 above)
Comment (raahi-chada2004) [reply out_writeapi_test.go]: Done. I also deleted the log file so you don't see anything. I'm not sure how else to omit the changes.

File: out_writeapi_test.go [hunk 2]
```diff
@@ -38,3 +56,136 @@ func TestFLBPluginRegister(t *testing.T) {
...
+	})
+	defer patch1.Unpatch()
+
+	patch2 := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		currChecks.calledSetContext++
+	})
+	defer patch2.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	initsize := len(configMap)
```
Comment (agrawal-siddharth) [inline out_writeapi_test.go]: It feels odd to be accessing an internal member variable used by the plugin. Can you add a test-only method on the plugin that provides the instance count and use that instead?

**Review by agrawal-siddharth — CHANGES_REQUESTED** [review]

File: out_writeapi_test.go (hunk 2 above)
Comment (raahi-chada2004) [reply out_writeapi_test.go]: Done.

**Review by agrawal-siddharth — APPROVED** [review]

**Pull request merged**